File: app/src/seller/java/sembako/sayunara/adapter/UserAdapter.java
package sembako.sayunara.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;

import sembako.sayunara.android.R;
import sembako.sayunara.android.ui.component.account.login.data.model.User;
import sembako.sayunara.android.ui.util.HFRecyclerViewAdapter;
import sembako.sayunara.android.ui.util.TimeUtils;


public class UserAdapter extends HFRecyclerViewAdapter<User, UserAdapter.ItemViewHolder> {

    private OnItemClickListener onItemClickListener;
    int minteger = 0;
    protected int total = 0;

    Context context;
    interface OnItemSelected {
        void onSelect(User model);
    }

    private OnItemSelected listener;


    public UserAdapter(Context context1) {
        super(context1);
        context = context1;
    }

    @Override
    public void footerOnVisibleItem() {

    }


    @Override
    public ItemViewHolder onCreateDataItemViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_user, parent, false);
        return new ItemViewHolder(v);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindDataItemViewHolder(ItemViewHolder holder, final int position) {
        User user = getData().get(position);
        holder.tvUsername.setText(user.getProfile().getUsername());
        holder.tvJoinDate.setText("Bergabung "+new TimeUtils().formatDateToIndonesia(user.getCreatedAt().getIso()));
        holder.tvNumber.setText(String.valueOf(position+1));
    }

    class ItemViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {


        TextView tvUsername,tvJoinDate,tvNumber;

        public ItemViewHolder(View view) {
            super(view);
            tvUsername   =  view.findViewById(R.id.tv_username);
            tvJoinDate  = view.findViewById(R.id.tv_join);
            tvNumber = view.findViewById(R.id.tv_number);
        }

        @Override
        public void onClick(View v) {

        }
    }
    public void actionDetail(OnItemClickListener actionQuestion) {
        onItemClickListener = actionQuestion;
    }public interface OnItemClickListener {
        void OnActionClickQuestion(View view, int position);
    }

}

File: app/src/seller/java/sembako/sayunara/adapter/UserAdapter.kt
package sembako.sayunara.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import sembako.sayunara.android.R
import sembako.sayunara.android.ui.component.account.login.data.model.User
import sembako.sayunara.android.ui.util.HFRecyclerViewAdapter
import sembako.sayunara.android.ui.util.TimeUtils
import java.util.ArrayList

class UserAdapter(var context: Context, onClickListener: OnClickListener) : HFRecyclerViewAdapter<User?, UserAdapter.ItemViewHolder?>(context) {

    private var mOnClickListener: OnClickListener
    private var resultList: List<User> = ArrayList()


    init {
        this.mOnClickListener = onClickListener
    }

    override fun footerOnVisibleItem() {}
    override fun onCreateDataItemViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.list_item_user, parent, false)
        return ItemViewHolder(v)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindDataItemViewHolder(holder: ItemViewHolder?, position: Int) {
        val user = data[position]
        holder?.tvUsername?.text = user?.profile?.username
        holder?.tvJoinDate?.text = "Bergabung " + TimeUtils().formatDateToIndonesia(user?.createdAt?.iso)
        holder?.tvNumber?.text = (position + 1).toString()

        if(user?.profile?.verified==true){
            holder?.ivVerified?.visibility = View.VISIBLE
        }else{
            holder?.ivVerified?.visibility = View.GONE
        }

        holder?.rlDetail?.setOnClickListener {
            //Toast.makeText(context,"gagaggaga",Toast.LENGTH_LONG).show()
            mOnClickListener.onClickDetail(position,user!!)
        }
    }

    inner class ItemViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var tvUsername: TextView = view.findViewById(R.id.tvUsername)
        var tvJoinDate: TextView = view.findViewById(R.id.tvJoinDate)
        var tvNumber: TextView = view.findViewById(R.id.tvNumber)
        var ivVerified : ImageView = view.findViewById(R.id.ivVerified)
        var rlDetail : RelativeLayout = view.findViewById(R.id.rlDetail)


    }

    fun setItems(resultList: List<User>, onClickListener: OnClickListener) {
        this.resultList = resultList
        this.mOnClickListener = onClickListener
        notifyDataSetChanged()
    }

    interface OnClickListener {
        fun onClickDetail(position: Int,user: User)
    }


}